packages/realm/src/errors.ts

```diff
@@ -138,3 +140,45 @@ export class ClientResetError extends SyncError {
     };
   }
 }
+
+/**
+ * An error class that indicates that one or more object changes have been reverted by the server.
+ * This can happen when the client creates/updates objects that do not match any subscription, or performs writes on
+ * an object it didn't have permission to access.
+ */
+export class CompensatingWriteError extends SyncError {
+  /**
+   * The array of compensating writes performed by the server.
+   */
+  public infos: CompensatingWriteInfo[] = [];
+
+  /** @internal */
+  constructor(error: binding.SyncError) {
+    super(error);
+    if (error.compensatingWritesInfo) {
+      error.compensatingWritesInfo.forEach((element) => {
+        this.infos.push({ objectName: element.objectName, reason: element.reason, primaryKey: element.primaryKey });
+      });
```

Comment:
Not sure if you actually have to create a separate object. If not: 
```ts
for (const write of error.compensatingWritesInfo) {
  // IF RENAMING `infos` TO `writes`:
  this.writes.push(write);
}
```

Otherwise:
```ts
for (const write of error.compensatingWritesInfo) {
  // IF RENAMING `infos` TO `writes`:
  this.writes.push({ ...write });
}
```

P.s. `forEach()` is quite slow.

Reply:
Makes sense! And good to know about foreach 😁